Resources/Prototypes/EntityTables/security.yml

```diff
@@ -0,0 +1,71 @@
+- type: weightedRandomEntity
+  id: BasicLootSecurityEntityTable
+  weights:
+    Flash: 9
+    Handcuffs: 9
+    Zipties: 9
+    FlashlightSeclite: 9
+    Tourniquet: 9
+    ClothingEyesGlassesSunglasses: 1
+    BoxHandcuff: 1
+
+- type: weightedRandomEntity
+  id: LootSecurityEntityTable
+  weights:
+    Flash: 9
+    Handcuffs: 9
+    Zipties: 9
+    FlashlightSeclite: 9
+    GrenadeFlashBang: 9
+    SecurityWhistle: 9
+    Tourniquet: 9
+    ClothingEyesGlassesSunglasses: 1
+    BoxHandcuff: 1
+    BoxFlashbang: 1
+    TearGasGrenade: 1
+    WeaponDisabler: 1
+    Stunbaton: 1
+
+- type: weightedRandomEntity
+  id: LootArmoryEntityTable
+  weights:
+    RiotShield: 95
+    RiotBulletShield: 95
+    RiotLaserShield: 95
+    ClothingHeadHelmetBasic: 95
+    ClothingHeadHelmetRiot: 95
+    ClothingOuterArmorBasic: 95
+    ClothingOuterArmorRiot: 95
+    ClothingOuterArmorBulletproof: 95
+    WeaponDisabler: 95
+    WeaponPistolMk58: 95
+    WeaponRifleLecter: 95
+    WeaponSubMachineGunDrozd: 95
+    WeaponLaserCarbine: 95
+    WeaponShotgunKammerer: 95
+    ClothingHeadHelmetSwat: 5
+    WeaponShotgunEnforcer: 5
```

Comment:
So, previously, it was a 95% chance to pick anything from the basic table and a 5% chance to pick from the rare stuff.

The total weights of the basic stuff add up to 1330 while the basic stuff adds up to 10.

10 / 1340 = 0.746% chance to grab anything from the rare table with a 99.3% chance to grab anything from the common table.

I don't think this weight system is an equal implementation to how the choices were done previously, let me know if my math is wrong.

Reply:
> I don't think this weight system is an equal implementation to how the choices were done previously, let me know if my math is wrong.

It isn't, because it's not really possible to do with weights alone.
The way it is now is like that - it uses 2 separate lists of entity prototypes, with each list having a chance to roll. Then, a random prototype is picked from the list. Basically, this is a mix of a chance roll and a random pick from a list. My system simply uses a single weighted table.

If needed, I can adjust the weights in the entity tables, since those can definitely be inaccurate relative to the old ones.

Reply:
How about instead of a single EntityTable it uses a `Dictionary<ProtoId<WeightedRandomEntityPrototype>, float> EntityTables` with the float being the weighted value chance of picking that table so you can use `_robustRandom.Pick(EntityTables)` on it

Reply:
This is basically the only reasonable way to maintain the set percentage without adjusting the weights for each item every single time someone adds another item

Reply:
alright, got it